Breeding-Insight/bi-web
# BI-1721 - Show All - Breeding Methods table

# Description
**Story:** https://breedinginsight.atlassian.net/browse/BI-1721

Original bug was that the "Show All" button wasn't working for the Breeding Methods table.  When investigating how to fix this, there was a lot of duplicate and confusing code around pagination, so I refactored that to utilize `PaginationController` and the mechanism for a parent component to update page information as needed, and for the `ExpandableTable` component to now be primarily responsible for handling paging events from the `PaginationControls`.



# Dependencies
bi-api - [bug/BI-1721](https://github.com/Breeding-Insight/bi-api/pull/242)

# Testing
1. On a given table, click the 'Show All' button, and ensure that all records are now visible
2. Click the 'Show All' button again, and ensure that the table is paginated again
3. Ensure pagination works

Testing applies to the all tables:
- Admin Users
- Admin Programs
- Program Users
- Program Locations
- Program Breeding Methods
- Enabling System Breeding Methods
- All Germplasm
- Germplasm Lists
- Germplasm List Details
- Ontology (active)
- Ontology (archived)
- Experiments
- Import Experiment
- Import Ontology
- Import Germplasm


# Checklist:

- [x] I have run TAF: https://github.com/Breeding-Insight/taf/actions/runs/4297630354
